plugin/math-katex/math-katex.js

```diff
@@ -0,0 +1,523 @@
+	 */
+	var replaceFormulaTex = (function() {
+
+		var regexInline   = /([^\\])\$([^\$]*)\$/g;       // $ … $
+		var regexDisplay  = /([^\\])\$\$([^\$]*)\$\$/g;   // $$ … $$
+
+		// For workarounds:
+		var regexCssClass = /<span class="katex">/;       // Used to add classes
+
+
+		return function ( mode, text ) {
+			var regex;
+
+			if ( mode === 'inline'  ) {
+				regex = regexInline;
+			}
+			else if ( mode === 'display' ) {
+				regex = regexDisplay;
+			}
+			else {
+				throw new Error( 'Invalid mode: ' + mode );
+			}
+
+
+			function replacer( _, lookbehind, group, offset ) {
+				// (Depends on mode, thus nested.)
+
+				if ( mode === 'display' ) {
+					var prefix = '\\displaystyle {';
+					offset += 2;    // $$
+					offset -= prefix.length;
+
+					group = prefix + group + '}';
+				}
+				else {
+					offset++;    // $
+				}
+
+				var markup;
+
+				try {
+					markup = lookbehind + katex.renderToString( group );
+				}
+				catch ( error ) {
+					correctErrorPosition( error, offset );  // mutates the error
+					throw error;
+				}
+
+
+				// Add a class for the mode to the root element
+				markup = markup.replace(
+					regexCssClass,
+					'<span class="katex ' + mode + '">'
+				);
+
+				return markup;
+			}
+
+			text = preProcess( text );
+			return text.replace( regex, replacer );
+		};
+
+	})();
+
+
+	/**
+	 * Unescapes `\$` to `$` (for `tex` mode).
+	 */
+	var unescapeDollarSign = (function() {
+
+		var regexUnescape = /\\\$/g;
+
+		return function ( text ) {
+			return text.replace( regexUnescape, '$' );
+		};
+	})();
+
+
+
+	// --- Formula error handling ----------------------------------------------
+
+	/**
+	 * Change the `position` property of an error object thrown by KaTeX to get
+	 * a more accurate error position. Mutates the passed object.
+	 */
+	function correctErrorPosition( error, offset ) {
+		offset = offset || 0;
+		error.position += offset;
+
+		// Look for "… got '…' …" error message to adjust the position
+		// (this will probably be broken by KaTeX updates).
+
+		var matches = error.message.match( /got\s'(.*?)'/ );
+		if ( matches !== null ) {
+			// Text: "… got '…' …"
+			error.position -= matches[1].length;
+		}
+	}
+
+
+	/**
+	 * Handles a KaTeX error by showing a more meaningful error message and
+	 * highlighting the error on the slide.
+	 *
+	 * Wraps the error with a `span` with class `math-katex-error`. Notifies the
+	 * user, but only for the first error.
+	 */
+	var handleError = (function() {
+
+		var showedError = false;
+
+		return function( error, formulaElement ) {
+
+			var e = formulaElement;
+			var slideNumber = getSlideNumber( e );
+
+			// Wrap
+			var s = e.innerHTML;
+			e.innerHTML = s.slice( 0, error.position ) +
+			              '<span class="math-katex-error">' +
+			              s.slice( error.position ) +
+			              "</span>";
+
+			// Log
+			console.error( 'Formula error on slide ' + slideNumber, error );
+
+			// Just show a `window.alert`
```

Comment:
Nope, it *should* disrupt, otherwise you won't notice errors while editing. It will however only show one alert, for the first error.